### PR TITLE
fix: remove invalid es-builtins

### LINKED ISSUE #260: Bug: no-unsupported-features/es-builtins false positives
### Environment

Node version:v22.0.0
npm version:v10.4.0
ESLint version:v9.0.0
eslint-plugin-n version: 17.3.1
Operating System:darwin 23.5.0

### What rule do you want to report?

no-unsupported-features/es-builtins

### Link to Minimal Reproducible Example

n/w

### What did you expect to happen?

```js
'use strict';
fetch('/api/xxx');
```

result:
```bash
  2:1  error  The 'fetch' is still an experimental feature and is not supported until Node.js 18.0.0. The configured version range is '>=17'  n/no-unsupported-features/es-builtins
```

Since `fetch` is not an es builtin, the rule should not report this case.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by aladdin-add**
These globals don't belong to ecma262, I moved them to node builtins. :)

fixes #260

### CODE REVIEW

**Comment by aladdin-add** [comment]
It may introduce new linting errors, which, according to eslint's versioning policy, I categorize as feat.

https://github.com/eslint/eslint?tab=readme-ov-file#semantic-versioning-policy

File: lib/rules/no-unsupported-features/es-builtins.js [hunk 1]
```diff
@@ -419,203 +419,6 @@ const traceMap = {
         unescape: {
             [READ]: { supported: ["0.10.0"] },
         },
-
```
Comment (scagood) [inline lib/rules/no-unsupported-features/es-builtins.js:422]: I think these should remain here 👀

File: lib/rules/no-unsupported-features/node-builtins.js [hunk 2]
```diff
@@ -31,6 +31,200 @@ const traceMap = {
                 paths: { [READ]: { supported: ["8.9.0"] } },
             },
         },
+
+        // General APIs
```
Comment (scagood) [inline lib/rules/no-unsupported-features/node-builtins.js:35]: I wanted to make a custom list from the node globals
https://nodejs.org/api/globals.html (via https://nodejs.org/api/globals.json)

As the numbers are quite different between mdn and the node docs 

If I find the time I will see if I can make a script to scrape that data, but so far I have found the scripts to be lacking 😅

**Review by scagood — CHANGES_REQUESTED** [review]
I am not sure this is what we actually want to do 🤔

**Comment by aladdin-add** [comment]
it's to fix the issue: https://github.com/eslint-community/eslint-plugin-n/issues/260

**Review by scagood — APPROVED** [review]
I think this resolves #260 well

**Comment by aladdin-add** [comment]
tips: it's recommended to use rebase, to help release-please generate better changelogs. :)

**Comment by scagood** [comment]
As we're using squash merge, it actually makes no difference, as only the PR title is kept 👀

If we're using merge / rebase merge then 100% we'd need to worry :)

**Comment by aladdin-add** [comment]
"node builtins" is repeated 2 times, and I'm not sure if it has anything to do with the use of merge:
https://github.com/eslint-community/eslint-plugin-n/blob/a70789a90c118f1ab69b8e1f5d4f1ac7f5dc3869/CHANGELOG.md#-features

**Comment by scagood** [comment]
😱 that makes no sense. I guess I will steer clear for now!

**Pull request merged**